openshift-kube-apiserver/admission/autoscaling/managementcpusoverride/admission.go

```diff
@@ -0,0 +1,551 @@
+}
+
+func (a *managementCPUsOverride) getPodNamespace(attr admission.Attributes) (*corev1.Namespace, error) {
+	ns, err := a.nsLister.Get(attr.GetNamespace())
+	if err == nil {
+		return ns, nil
+	}
+
+	if !errors.IsNotFound(err) {
+		return nil, admission.NewForbidden(attr, err)
+	}
+
+	// in case of latency in our caches, make a call direct to storage to verify that it truly exists or not
+	ns, err = a.client.CoreV1().Namespaces().Get(context.TODO(), attr.GetNamespace(), metav1.GetOptions{})
+	if err == nil {
+		return ns, nil
+	}
+
+	if !errors.IsNotFound(err) {
+		return nil, admission.NewForbidden(attr, err)
+	}
+
+	return nil, err
+}
+
+func (a *managementCPUsOverride) waitForSyncedStore(timeout <-chan time.Time) bool {
+	for !a.nsListerSynced() || !a.nodeListSynced() {
+		select {
+		case <-time.After(100 * time.Millisecond):
+		case <-timeout:
+			return a.nsListerSynced() && a.nodeListSynced()
+		}
+	}
+
+	return true
+}
+
+func updatePodResources(pod *coreapi.Pod, workloadType string, class coreapi.PodQOSClass) {
+	if pod.Annotations == nil {
+		pod.Annotations = map[string]string{}
+	}
+
+	// update init containers resources
+	updateContainersResources(pod.Spec.InitContainers, pod.Annotations, workloadType, class)
+
+	// update app containers resources
+	updateContainersResources(pod.Spec.Containers, pod.Annotations, workloadType, class)
+
+	// re-add workload annotation
+	addWorkloadAnnotations(pod.Annotations, workloadType)
+}
+
+func updateContainersResources(containers []coreapi.Container, podAnnotations map[string]string, workloadType string, podQoSClass coreapi.PodQOSClass) {
+	for i := range containers {
+		c := &containers[i]
+		cpusharesAnnotationKey := fmt.Sprintf("%s.%s.cpushares/%s", containerResourcesAnnotationPrefix, workloadType, c.Name)
+
+		// make sure best effort is always 2 shares, it the minimal shares that supported
+		// see - https://github.com/kubernetes/kubernetes/blob/46563b0abebbb00e21db967950a1343e83a0c6a2/pkg/kubelet/cm/qos_container_manager_linux.go#L99
+		if podQoSClass == coreapi.PodQOSBestEffort {
+			podAnnotations[cpusharesAnnotationKey] = "2"
+			continue
+		}
+
+		if c.Resources.Requests != nil {
+			// if the kubelet deleted container CPUs requests from resources for static pods,
+			// the admission plugin will not do additional updates
+			if _, ok := c.Resources.Requests[coreapi.ResourceCPU]; !ok {
+				continue
+			}
+
+			cpuRequest := c.Resources.Requests[coreapi.ResourceCPU]
+			cpuRequestInMilli := cpuRequest.MilliValue()
+
+			cpuShares := cm.MilliCPUToShares(cpuRequestInMilli)
+			podAnnotations[cpusharesAnnotationKey] = fmt.Sprintf("%d", cpuShares)
+			delete(c.Resources.Requests, coreapi.ResourceCPU)
+
+			if c.Resources.Limits == nil {
+				c.Resources.Limits = coreapi.ResourceList{}
+			}
+
+			// multiply the CPU request by 1000, to make sure that the resource will pass integer validation
+			managedResource := fmt.Sprintf("%s.%scores", workloadType, podWorkloadAnnotationPrefix)
```

Comment:
interestingly, this prefix isn't a prefix.  

I prefer prefixes to suffixes.  Let's make a complete prefix for what we want.

Reply:
In general, the managed resource has a form
<workload_type>.workload.openshift.io/cores
I created the constant 
```
// containerWorkloadResourceSuffix contains the suffix for the container workload resource
containerWorkloadResourceSuffix = "workload.openshift.io/cores"
```
that I will use here